runtime/pkg/dag/dag_test.go

```diff
@@ -82,3 +85,34 @@ func getTestDAG() *DAG {
 	//     B2  A2
 	return d
 }
+
+func TestCyclicDAG(t *testing.T) {
+	d := NewDAG()
+	d.Add("A0", []string{"B1"})
+	n, err := d.Add("B1", []string{"A0"})
+	require.Error(t, err)
+
+	d.Delete("A0")
+	d.Delete("B1")
+	d.Add("A0", []string{})
+	d.Add("B1", []string{"A0"})
+	d.Add("B2", []string{"B1"})
+	n, err = d.Add("B0", []string{"B1"})
```

Comment:
Please check errors for all adds to make test case safer

Reply:
Added for all the `Add` calls in all the tests.